### DISCUSSION TITLE
AWS SSM RDP support for remote sessions

### DISCUSSION THREAD

**Discussion by carlfugate** [Noovolari/leapp#401] [Ideas]
It would be great if the connection options for AWS SSM enabled instances supported Remote Desktop Protocol (RDP).  This is done by using local port forwarding via the SSM agent.  It would require dynamic selection of an available 'localPortNumber' to map to the SSM session.

Command: aws ssm start-session --target <instance-id> --document-name AWS-StartPortForwardingSession --parameters "localPortNumber=55678,portNumber=3389"

Once this is complete similar to the launch of other terminal sessions could launch the RDP application of choice (using rdp:// or passing parameters to the app)

Example
RDP via Session Manager: https://awscloudsecvirtualevent.com/workshops/module1/rdp/

**Comment by andreacavagna01** [comment]
Great Idea! I also move the conversation to the Slack community! I think we have a community member that started implementing a solution for that

**Comment by andreacavagna01** [comment]
For a quick answer, a community member created a plugin for enhanced AWS SSM parameters:
https://github.com/christian-calabrese/leapp-ssm-tunnels-plugin.

You can install it from https://www.leapp.cloud/plugins

But I completely agree, it deserve a better integration
